app/services/book_best_sellers_service.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class BookBestSellersService
```

Comment:
It should query object, also query object MUST return ASSOCIATION. Check your code

<img width="794" alt="image" src="https://user-images.githubusercontent.com/62115862/195625834-4f446a83-dd8e-4fd2-943a-e4b0f4047815.png">


And use active record SQL logic, do not use map/sort/uniq etc for such queries, its very load
For best practices, write one big query or use small queries. DONT use array/enumeable methods for such queries